confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java

```diff
@@ -202,9 +205,12 @@ public ConfigManager() throws IOException {
     this.loadManager = new LoadManager(this);
     this.syncManager = new SyncManager(this, syncInfo);
 
-    // ConsensusManager must be initialized last, as it would load states from disk and reinitialize
-    // above managers
-    this.consensusManager = new ConsensusManager(this, stateMachine);
+    // ConsensusManager doesn't initialize until the ConfigNode knows its own nodeId
+    this.consensusManager = null;
+  }
+
+  public void initConsensusManager() throws IOException {
+    this.consensusManager = new ConsensusManager(this, this.stateMachine);
```

Comment:
Please add check of whether consensusManager is already initialized. if this method is called be a Thread-Safe method, then you can simply judge whether consensusManager is null. Or else you need to guarantee consensusManager only be initialized one time.